src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    if (this[0] !== null) {
+      result += `${this[0]}`;
+    }
+
+    for (let i = 1; i < this.length; i++) {
+      if (this[i] !== undefined && this[i] !== null) {
+        result += `${separator}${this[i]}`;
+      } else {
+        result += `${separator}`;
+      }
```

Comment:
so, you add separator at any condition, right? then you must do it at the start of your loop, without condition block, to not repeat yourself twice and then you wouldn't need any `else` statement